lib/binary_tree.c

```diff
@@ -56,3 +56,22 @@ struct bt_node *bst_lookup(struct bt_node *root, int val)
         }
     }
 }
+
+int print_leaf_nodes(struct bt_node *node)
+{
+    if (node == NULL) {
+        fprintf(stderr, "%s: first argument 'node' is NULL\n", __func__);
```

Comment:
replaced with a warnining:
fprintf(stdout, "%s: Warn: first argument 'node' is NULL\n", __func__);